client/src/components/Tickets/TicketModal.js

```diff
@@ -221,16 +221,34 @@ const TicketHistoryCard = ({
   return (
     <Card className="mt-1">
       <CardHeader
+        tabIndex="0"
         onClick={() =>
           dispatch({
             type: 'TICKET_DETAILED_VIEW_TOGGLE_SHOW_ITEM',
             payload: { index: historyId }
           })
         }
+        onKeyDown={useCallback(
+          e =>
+            e.key === ' ' &&
+            dispatch(
+              {
+                type: 'TICKET_DETAILED_VIEW_TOGGLE_SHOW_ITEM',
+                payload: { index: historyId }
+              },
+              e.preventDefault()
+            )
+        )}
       >
         <span className="ticket-history-header d-inline-block text-truncate">
           <strong>
-            <span className={ticketHeaderClassName}>
+            <span
+              className={ticketHeaderClassName}
+              id="TicketHeader"
+              role="button"
+              aria-expanded="false"
```

Comment:
__Concern__
The code does not appear to update the value of this attribute, but this attribute is expected to change dynamically between `false` and `true` depending on current state of the content expansion.

<details>
<summary>Reference</summary>

- [example code (with explanation)](https://www.w3.org/WAI/GL/wiki/Using_the_WAI-ARIA_aria-expanded_state_to_mark_expandable_and_collapsible_regions#Examples)
- [live example (with explanation)](http://3needs.org/en/testing/code/aria-expanded.html)
- [live example (sans explanation)](http://3needs.org/en/testing/code/aria-expanded-code.html)

([source of content](http://3needs.org/en/testing/sr-aria.html))

</details>

Reply:
@tacc-wbomar, I refactored the code to allow the `aria-expanded` attribute to dynamically change. Everything _should_  still work the same (or at least it did on my end), but now `aria-expanded` should properly update.

List of changes: 
- Since the last ticket history card is always open, but `aria-expanded` was set to "false" by default, I changed this to check whether each card `isOpen` to determined the `aria-expanded` value
  - Now all previous closed cards should have `aria-expanded="false"` while the most recent opened card has `aria-expanded="true"`
- Pulled `onClick()` and `onKeyDown()` out of the `<CardHeader>` HTML and set them as functions above
- Set a `ticketHeaderRef` in the component definition so that `aria-expanded` could be referenced and updated upon opening a history card body

https://user-images.githubusercontent.com/51714485/145285691-4ff17bd4-29f6-4d26-b433-8779b00e8685.mov

Reply:
The `ticketHeaderRef` is a solution (as you've proven), but it does two concerning things:
- mimics(?) the state of `<Collapse>`
- manipulates DOM[^1]

[^1]: JS frameworks, like React, are intended to handle this for us by binding JS variables, like `isOpen` to attributes.

Can we rely on `<Collapse>`'s internal state? (See [reactstrap > Collapse > Events](https://reactstrap.github.io/?path=/docs/components-collapse--collapse-events#collapse-events).)

<details><summary>Ideas</summary>

Run code on `<Collapse>`'s `onEntered` and `onExited` function props to…

(Tracking state is **not** my specialty. I just smell a cleaner way.)

</details>

Reply:
@duckonomy Ahhh, that makes sense. I did run into the problem of string vs. boolean when adding the ref and thought I had gotten it right, but apparently not. I removed everything related to the ticketHeaderRef. I also changed the ternary in `aria-expanded` to just `{isOpen}` like you suggested. I don't know why I didn't realize `isOpen` already contained the value I needed... thank you for the pointers here!

@tacc-wbomar Now that the ticketHeaderRef `onKeyDown()` code has been removed, the DOM manipulation and `<Collapse>` mimicking issues should be resolved right?

Reply:
Yup. Thanks y'all.